packages/loader/driver-utils/src/treeUtils.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    SummaryType,
+    ISummaryTree,
+    SummaryObject,
+} from "@fluidframework/protocol-definitions";
+
+/**
+ * Summary tree assembler without stats collection.
+ */
+export class SummaryTreeAssembler {
```

Comment:
I think this file / class should move to the driver layer (routerlicious-driver) for couple of reasons:
1. back-compat concerns - The loader can be on an older version and if so, the wholeDocumentStorageService won't find this.
2. This is only used by routerlicious-driver so we should probably keep it there for now. If this becomes more generic, we should move it to a generic util.

Reply:
Here are some thoughts here:

1. driver-utils seem to be consumed at a build time. Meaning the utilities are not consumed through "definitions" boundary (I don't think layer checks would let us consume in driver -> loader direction anyways). So may not have runtime concerns here. Maybe driver-utils should be located within `packages/drivers` (lets adapt that through another PR if needed). It seems that idea is similar to container-utils that state "Contains various helper utilities available to both runtime & container layers".  Similarly, driver utilities should service driver and layers above it.
2. We discussed earlier about making sure that we are creating re-usable set of conversion functions.  This particular conversion is not specific to Routerlicious in terms or types it consumes. If we place it in routerlicious driver, we are creating a friction (and probably developers' question why it is there), around later adoption. It may make more sense to keep it here, so if future changes are needed, developer would be clear they need to modify this functionality as it has more generic purpose.